prf/hmac-sha256/src/leader.rs

```diff
@@ -0,0 +1,208 @@
+use std::{marker::PhantomData, sync::Arc};
+
+use async_trait::async_trait;
+use futures::lock::Mutex;
+
+use hmac_sha256_core::{
+    self as leader_core, PRFLeaderConfig, PmsLabels, SessionKeyLabels, MS, SESSION_KEYS,
+};
+use leader_core::{CF_VD, SF_VD};
+use mpc_aio::protocol::garble::{exec::dual::DEExecute, factory::GCFactoryError};
+use mpc_core::garble::{
+    exec::dual::{DualExConfig, DualExConfigBuilder},
+    ChaChaEncoder,
+};
+use utils_aio::factory::AsyncFactory;
+
+use crate::{circuits, PRFLeader, State};
+
+use super::PRFError;
+
+pub struct DEPRFLeader<DEF, DE>
```

Comment:
Can we change this name to `PRFLeader`? Same for follower.

Reply:
We could name the trait `PRFLead`?  Only asked because it will be a pain to type this :smile:

Something different: I encountered the pattern of naming the trait exactly as the struct/implementer several times. I think this is a bad idea, because it leads to more confusion especially for people not familiar with the code.